testsuite/MDAnalysisTests/core/test_topologyattrs.py

```diff
@@ -501,3 +501,15 @@ def test_static_typing_from_empty():
 
     assert isinstance(u._topology.masses.values, np.ndarray)
     assert isinstance(u.atoms[0].mass, float)
+
+
+def test_warn_selection_for_strange_dtype():
+    with pytest.warns(UserWarning) as rec:
+        class Star(tpattrs.TopologyAttr):
+            singular = "star"  # turns out test_imports doesn't like emoji 
+            attrname = "stars"  # :(
+            per_object = "atom"
+            dtype = dict
+
+    err = "A selection keyword could not be automatically generated"
+    assert err in rec[0].message.args[0]
```

Comment:
You can make the test more concise by using https://docs.pytest.org/en/stable/warnings.html
```python
with pytest.warns(..., match=err):
    ...
```